### ISSUE TITLE
Training results in NaN (commit 2b03725)

### ISSUE THREAD

**Issue by bliep** [openai/pixel-cnn#27]
Running `train.py --nr_gpu=1` results in NaN for the loss`bits_per_dim` on tensorflow 1.5.0.
With some API changes (`keepdims` -> `keep_dims`, `axis=` -> '') the same NaN occurs on tensorflow 1.4.x.
Disabling the data driven initialization of the weight normalization (commenting out `sess.run(init_pass, feed_dict)`) however does not results in NaN loss.

**Comment by yang-song** [comment]
I have the same problem.

**Comment by learning-luke** [comment]
I'm also getting an error like this. What did you change `axis=` into, by the way? I changed it to `dim=` to get it to work on tf 1.4.

**Comment by attaullah** [comment]
same issue here.
